### PR TITLE
[Enhancement] Support for setting transaction thread pool properties.

### LINKED ISSUE #50091: Threads in the publish_version/update_apply thread pool are created frequently.
## Enhancement
Frequent thread creation imposes a performance overhead
企业微信截图_bc3d9452-dab3-4dff-88bb-f30aef5d3ea1
企业微信截图_c58e2d7e-9d04-40e9-9ba5-9a2e6f736575
企业微信截图_0d093e22-2864-453f-9a7d-7a9752a44408
企业微信截图_25189d0e-25b5-48f1-90a0-cd226e01846b

### PR DESCRIPTION
**Pull request by zhangheihei**
## Why I'm doing:
Support for setting transaction thread pool properties.
## What I'm doing:

Fixes #50091

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -95,11 +95,15 @@ CONF_Int32(push_worker_count_high_priority, "3");
 
 // The count of thread to publish version per transaction
 CONF_mInt32(transaction_publish_version_worker_count, "0");
+// The min count of thread to publish version per transaction
+CONF_Int32(transaction_publish_version_worker_count_min, "0");
```
Comment (kevincai) [inline be/src/common/config.h:99]: make it dynamic mutable whenever possible.

And how this configuration differentiate to `transaction_publish_version_worker_count`?

File: be/src/common/config.h [hunk 2]
```diff
@@ -95,11 +95,15 @@ CONF_Int32(push_worker_count_high_priority, "3");
...
 
 // The count of thread to apply rowset in primary key table
 // 0 means apply worker count is equal to cpu core count
 CONF_mInt32(transaction_apply_worker_count, "0");
 CONF_mInt32(get_pindex_worker_count, "0");
+CONF_Int32(transaction_apply_worker_count_min, "0");
```
Comment (kevincai) [inline be/src/common/config.h:105]: similar question, how does this configvar differs to `transaction_apply_worker_count`, and when `transaction_apply_worker_count` set to `0`, it means use the number of cores, will it apply to `transaction_apply_worker_count_min` as well?

make it dynamic adjustable if possible.

Comment (zhangheihei) [reply be/src/common/config.h:105]: I'll adjust the change attribute to dynamically update

**Review by trueeyu — DISMISSED** [review]

File: be/src/http/action/update_config_action.cpp [hunk 3]
```diff
@@ -165,6 +165,10 @@ Status UpdateConfigAction::update_config(const std::string& name, const std::str
             (void)thread_pool->update_max_threads(
                     std::max(MIN_TRANSACTION_PUBLISH_WORKER_COUNT, config::transaction_publish_version_worker_count));
         });
+        _config_callback.emplace("transaction_publish_version_thread_pool_num_min", [&]() {
+            auto thread_pool = ExecEnv::GetInstance()->agent_server()->get_thread_pool(TTaskType::PUBLISH_VERSION);
+            (void)thread_pool->update_min_threads(config::transaction_publish_version_thread_pool_num_min);
```
Comment (wyb) [inline be/src/http/action/update_config_action.cpp:170]: should check `MIN_TRANSACTION_PUBLISH_WORKER_COUNT`

Comment (zhangheihei) [reply be/src/http/action/update_config_action.cpp]: thks,review. I had added the check.

**Review by wyb — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Pull request merged**